src/main/java/org/nastya/service/UserService.java

```diff
@@ -6,6 +6,7 @@
 import org.nastya.service.exception.UserAlreadyExistsException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpHeaders;
```

Comment:
Your services with business logic should not know anything about http or the way the are being invoked,

Reply:
Remove this class out of here.